tests/console/yast2_samba.pm

```diff
@@ -14,258 +14,352 @@ use strict;
-    assert_screen 'yast2_samba_open_port_firewall';
+    # service starts during boot, wait to load default data
+    assert_screen 'yast2_samba-startup-configuration';
+    wait_screen_change { send_key 'alt-r' };
+    assert_screen 'yast2_samba-server_start-during-boot';
+    # open firewalld port
     wait_screen_change { send_key 'alt-f' };
-    send_key 'alt-i';
-
-    assert_screen 'yast2_samba_firewall_port_details';
-    wait_screen_change { send_key 'alt-e' };
-    wait_screen_change { send_key 'alt-a' };
-    wait_screen_change { send_key 'alt-o' };
+    assert_screen 'yast2_samba_open_port_firewall';
 
-    # switch to Samba Configuration - Shares
-    send_key 'alt-s';
+    ## switch to Samba Configuration - Shares
+    wait_screen_change { send_key 'alt-s' };
+    wait_still_screen(2);
     assert_screen 'yast2_samba-server_shares';
-
     # add a shares config html_public
     wait_screen_change { send_key 'alt-a' };
     assert_screen 'yast2_samba-server_new-share';
-
-    type_string 'html_public';
+    # enter share details
+    type_string('html_public');
     wait_screen_change { send_key 'alt-a' };    # set share description
-    type_string 'html docs for share';
+    type_string($samba_directives->{comment});
     wait_screen_change { send_key 'alt-s' };
-    type_string '/html_public';                 # set share path to /home/html_public
+    for (1 .. 5) { send_key 'backspace'; }
+    type_string($samba_directives->{path});     # set share path to /home/html_public
     wait_screen_change { send_key 'alt-r' };    # set read-only
-
-    # check config before confirm new share with ok, confirm to create new share path
+                                                # check config before confirm new share with ok, confirm to create new share path
     assert_screen 'yast2_samba-server_new-share_create';
     wait_screen_change { send_key 'alt-o' };
     assert_screen 'yast2_samba-server_new-share-path';
-    wait_screen_change { send_key 'alt-y' };
-
-    # back to samba configuration and make some changes to share directories
+    wait_screen_change { send_key 'alt-y' };    # confirm new path
+                                                # back to samba configuration and make some changes to share directories
+    wait_still_screen(2);
     wait_screen_change { send_key 'alt-w' };    # allow users to share directories
     wait_screen_change { send_key 'alt-g' };    # allow guest access
     wait_screen_change { send_key 'alt-m' };
-    for (1 .. 10) { send_key 'down'; }          # Maximum number of shares
+    type_string_slow($samba_directives->{usershare_max_shares} . "\n");    # Maximum number of shares
 
-    # switch to identity configuration
+    ## switch to identity configuration
     send_key 'alt-d';
     assert_screen 'yast2_samba-server_identity';
 
+    ## domain controller in sle15
+    # select "Not a Domain Controller"
+    if ((is_sle(">=15")) || (is_leap(">=15.0"))) {
```

Comment:
also here as in a previous comment: Are you sure that TW is not behaving the same as SLE15 or leap15?